### ISSUE TITLE
Make `config` react to changes

### ISSUE THREAD

**Issue by chrisvxd** [measuredco/puck#689]
Currently, `config` doesn't react to changes. This makes certain behaviour impossible to implement, such as changing components at run-time.

## Considerations

* Will likely require a refactor of `createReducer`
* What happens to history if it references old components? Treat same as missing components?

**Comment by mauri-bask-health** [comment]
Hello! I was wondering if there are any updates or plans to work on this feature soon.

By the way, thank you for all the hard work you’re doing. Puck Editor is 🔥🔥 !!!

P.s. if I can be helpful in any way, please lmk

**Comment by chrisvxd** [comment]
Hey @mauri-bask-health, thanks for reaching out! It's not top of the queue, but it is on our radar.

Our priority features are #598 and #255. If there's enough interest in this one, we can bring it up the queue.

**Comment by chrisvxd** [comment]
Interested in doing this, but blocked by #644 which will update the state management.

**Comment by mindofjonas** [comment]
Any update here since #644 was closed?

**Comment by FedericoBonel** [comment]
Someone mentioned in the Discord server that it would be nice to have a `refreshConfig` function to trigger a "reaction" to the latest config programmatically.

**Comment by harbinsela** [comment]
I’m the one who originally opened the thread about this on Discord.

I’d like to provide a concrete real world use case. I’m building a multilingual page builder using Puck with runtime i18n. The page content updates language reactively without issues, but Puck’s config, including labels and built in field names like dimensions, width, text, numbers, strings do not react to language changes. From what I can tell, the config is consumed once at mount and then cached internally.

Recreating the config with useMemo does not help, and the only current workaround is remounting the entire editor using key={language}. That resets scroll position and selection state and negatively affects the user experience. Custom fields work because they are reactive React components, but built in field types do not.

I have tried multiple approaches, but nothing works without a full remount. The config is treated as static metadata rather than reactive state, and the TypeScript types are strictly defined, for example label is typed as string, which prevents passing functions or dynamic values. There is no supported way to update config dynamically once mounted.

The only theoretical workaround would be recreating built in field types such as arrays, selects, and strings as custom fields just to make them reactive. I have done this successfully for simple string fields and they work perfectly if you change the language they respond immediately because you wrap them as REACT ELEMENTS with type=custom, but doing the same for complex types like array would effectively mean re implementing core Puck field behavior, which is not practical. Arrays are not a single fixed UI. They can render in multiple ways depending on configuration, so wrapping them as a single custom field becomes unmanageable.

A refreshConfig API or making config reactive would solve this cleanly without forcing a full remount. This would significantly improve dynamic use cases like runtime i18n or feature toggling.
